src/js/components/Heading/__tests__/Heading-test.js

```diff
@@ -228,3 +228,13 @@ test('Throws a warning when heading.level is undefined in the theme.', () => {
   const consoleMsg = 'Heading level 6 is not defined in your theme.';
   expect(global.console.warn).toHaveBeenCalledWith(consoleMsg);
 });
+
+test('Heading fill renders', () => {
+  const component = renderer.create(
+    <Grommet>
+      <Heading fill />
```

Comment:
Not sure if I got it, but added an example using Box in storybook. Even tested them outside <Grommet> component, and they seem to have the same behavior.

```
<Grommet theme={grommet}>
      <Grid columns="large" gap="medium">
        <Set size="medium" />
        <Set size="small" />
        <Set size="large" />
        <Set size="xlarge" />
      </Grid>

      <Heading fill>Heading outside Box {headingFiller}</Heading>
      <Box>
        <Heading fill>Heading inside a Box {headingFiller}</Heading>
      </Box>
      <Box width="50%">
        <Heading fill>Heading inside a Box {headingFiller}</Heading>
      </Box>
    </Grommet>
```

Let me know if that's what you were thinking :smile: if it's not, we can try another storybook test.

Reply:
Can you confirm that these two children of the Grommet component are behaving exactly the same:
```
<Grommet theme={grommet}>
      <Heading fill>Heading outside Box {headingFiller}</Heading>
      <Box>
          <Heading fill>Heading inside a Box {headingFiller}</Heading>
      </Box>
</Grommet>
```
My thought is that the first Heading wouldn't behave the same as the one in the Box, since it doesn't have a layout wrapper, and 'fill' might not actually work for it. But I might be wrong. If both Heading work as expected then we are good. If the Heading that is wrapped with a Box is the only one that works as expected then we should change the test to use a Box wrapper.
For reference, See the [`margin` prop comment on docs](https://v2.grommet.io/text#margin):
```
...For vertical margin to be applied, Text needs to be contained within a layout component (such as Box or a generic div) or behave as a div (by applying as="div" or a display style of inline-block).
```
I'm not sure if that is exactly the case here, but I just want us to be sure the snapshots are portraying the right picture (same for Paragraph btw).

Reply:
Alongside storybook testing, did some tests using another project, without any CSS. Just the code you provided.

![image](https://user-images.githubusercontent.com/47114840/95671124-92f60380-0b69-11eb-9361-b5e6db3c0c46.png)

They are rendering with the same classes and no matter how I change the screen, the components always display the same styling and width/height.

The only thing I could point out is that sometimes when resizing the screen, Heading inside a Box breaks the line first, but they still show the same width when inspectioning. But as mentioned, that doesn't happen in every break. It'll depend on the  text used.

Added a background color to the headings manually:
![image](https://user-images.githubusercontent.com/47114840/95671239-bff6e600-0b6a-11eb-86b6-9fe927b04d47.png)

And even when they don't break lines together, they still have full width:

![image](https://user-images.githubusercontent.com/47114840/95671247-dd2bb480-0b6a-11eb-8a40-ebcf48744856.png)